pkg/healthcheck/healthcheck.go

```diff
@@ -1855,6 +1869,47 @@ func (hc *HealthChecker) CheckNamespace(ctx context.Context, namespace string, s
 	return nil
 }
 
+func (hc *HealthChecker) checkClusterNetworks(ctx context.Context) error {
+	nodes, err := hc.kubeAPI.GetNodes(ctx)
+	if err != nil {
+		return err
+	}
+	var podCIDRs []string
+	for _, node := range nodes {
+		podCIDRs = append(podCIDRs, node.Spec.PodCIDR)
+	}
+	clusterNetworks := strings.Split(hc.linkerdConfig.ClusterNetworks, ",")
+	clusterIPNets := make([]*net.IPNet, len(clusterNetworks))
+	for i, clusterNetwork := range clusterNetworks {
+		_, ipNet, err := net.ParseCIDR(clusterNetwork)
+		if err != nil {
+			return err
+		}
+		clusterIPNets[i] = ipNet
+	}
+	var badPodCIDRS []string
+	for _, podCIDR := range podCIDRs {
+		podIP, _, err := net.ParseCIDR(podCIDR)
+		if err != nil {
+			return err
+		}
+		exists := false
+		for _, clusterIPNet := range clusterIPNets {
+			if clusterIPNet.Contains(podIP) {
+				exists = true
+				break
+			}
+		}
+		if !exists {
+			badPodCIDRS = append(badPodCIDRS, podCIDR)
+		}
+	}
+	if len(badPodCIDRS) > 0 {
+		return fmt.Errorf("set the clusterNetworks config appropriately during install via linkerd --set clusterNetworks=%s", strings.Join(badPodCIDRS, ","))
```

Comment:
```suggestion
		return fmt.Errorf("set the clusterNetworks config appropriately during install via --set clusterNetworks=%s", strings.Join(badPodCIDRS, ","))
```

Reply:
I think it's more important that the check error message states the problem:
"node %s has a podCIDR (%s) which is not contained in the Linkerd clusterNetworks"

Advice for how to resolve the problem (setting the clusterNetworks value) can be either a second line of the error message or can be relegated to the help page linked to by the hint url.

Reply:
I lean towards @adleong's suggestion as well @alpeb